src/ParseNode.js

```diff
@@ -10,15 +13,15 @@ import type {Mode} from "./types";
  * For details on the corresponding properties see `Token` constructor.
  * Providing such information can lead to better error reporting.
  */
-export default class ParseNode {
-    type: *;
-    value: *;
+export default class ParseNode<TYPE: NodeType> {
```

Comment:
`ParseNode<T: NodeType>` is more common when defining parameterized types.

Reply:
I'm happy to change it if you insist. I just wanted to make sure here and in other places where I've used a parametrized type that it emphasizes that this is the `type` and not the `value` of the parse node. WDYT?

Reply:
I hadn't thought about that.  I guess we can leave it as is.